### ISSUE TITLE
Video player is no longer able to play videos

### ISSUE THREAD

**Issue by milano23** [flutter/flutter#43842]
My video player no longer plays any videos.  It did two days ago and as of yesterday it stopped.  I thought it was because i was transitioning some of my assets to firebase but that is not the case.  I am using appcenter to push the application to a couple of phones.  The phones have not had a new push in three days and the video player still works on them. 

## Steps to Reproduce
I have created a brand new project.  I have copied the video player example on [flutter site](https://flutter.dev/docs/cookbook/plugins/play-video).  I have tried this on android and ios.  On android simulator using vscode on a windows machine  i am receiving the following error:
```
The following assertion was thrown building FutureBuilder<void>(state:
I/flutter (13852): _FutureBuilderState<void>#26252):
I/flutter (13852): 'package:flutter/src/rendering/proxy_box.dart': Failed assertion: line 398 pos 15: 'aspectRatio >
I/flutter (13852): 0.0': is not true.
I/flutter (13852):
I/flutter (13852): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (13852): more information in this error message to help you determine and fix the underlying cause.
I/flutter (13852): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (13852):   https://github.com/flutter/flutter/issues/new?template=BUG.md
I/flutter (13852):
I/flutter (13852): Widget creation tracking is currently disabled. Enabling it enables improved error messages. It can
I/flutter (13852): be enabled by passing `--track-widget-creation` to `flutter run` or `flutter test`.
I/flutter (13852):
I/flutter (13852): When the exception was thrown, this was the stack:
I/flutter (13852): #2      new RenderAspectRatio (package:flutter/src/rendering/proxy_box.dart:398:15)
I/flutter (13852): #3      AspectRatio.createRenderObject (package:flutter/src/widgets/basic.dart:2673:65)
I/flutter (13852): #4      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4749:28)
I/flutter (13852): #5      SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5126:11)
I/flutter (13852): #6      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3101:14)
I/flutter (13852): #7      Element.updateChild (package:flutter/src/widgets/framework.dart:2904:12)
I/flutter (13852): #8      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3961:16)
I/flutter (13852): #9      Element.rebuild (package:flutter/src/widgets/framework.dart:3738:5)
I/flutter (13852): #10     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2348:33)
I/flutter (13852): #11     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:760:20)
I/flutter (13852): #12     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:280:5)
I/flutter (13852): #13     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1033:15)
I/flutter (13852): #14     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:975:9)
I/flutter (13852): #15     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:891:5)
I/flutter (13852): #19     _invoke (dart:ui/hooks.dart:249:10)
I/flutter (13852): #20     _drawFrame (dart:ui/hooks.dart:207:3)
I/flutter (13852): (elided 5 frames from class _AssertionError and package dart:async)
I/flutter (13852): 
I/flutter (13852): ════════════════════════════════════════════════════════════════════════════════════════════════════
```
On ios simulator  using vscode on a mac, the application crashed with the following error:
```Lost connection to device.
*** 
First throw call stack:
(
        0   CoreFoundation                      0x00007fff23baa1ee __exceptionPreprocess + 350
        1   libobjc.A.dylib                     0x00007fff50864b20 objc_exception_throw + 48
        2   Foundation                          0x00007fff256d3466 -[NSURL(NSURL) initFileURLWithPath:] + 575
        3   Foundation                          0x00007fff256d3c24 +[NSURL(NSURL) fileURLWithPath:] + 45
        4   Runner                              0x0000000106f2d03f -[FLTVideoPlayer initWithAsset:frameUpdater:] + 191
        5   Runner                              0x0000000106f3131d -[FLTVideoPlayerPlugin handleMethodCall:result:] + 1581
        6   Flutter                             0x0000000107cc04fd __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke + 104
        7   Flutter                             0x0000000107c59ec0 _ZNK7flutter21PlatformMessageRouter21HandlePlatform<…>
```
## Code
```
import 'package:flutter/material.dart';
import 'package:video_player/video_player.dart';

class VideoPlayerScreen extends StatefulWidget {
  @override
  _VideoPlayerScreen createState() => _VideoPlayerScreen();
}

class _VideoPlayerScreen extends State<VideoPlayerScreen> {
  VideoPlayerController _controller;
  Future<void> _initialVideoPlayer;

  @override
  void initState(){
    _controller = VideoPlayerController.network('https://flutter.github.io/assets-for-api-docs/assets/videos/butterfly.mp4');
    _initialVideoPlayer = _controller.initialize();
    super.initState();
  }

  @override
  void dispose(){
    _controller.dispose();

    super.dispose();
  }
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Video Player'),
      ),
      body: FutureBuilder(
        future: _initialVideoPlayer,
        builder: (context, snapshot){
          if(snapshot.connectionState == ConnectionState.done){
            return AspectRatio(
              aspectRatio: _controller.value.aspectRatio,
              child: VideoPlayer(_controller),
            );
          } else {
            return Center(child: CircularProgressIndicator());
          }
        },
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          // Wrap the play or pause in a call to `setState`. This ensures the
          // correct icon is shown.
          setState(() {
            // If the video is playing, pause it.
            if (_controller.value.isPlaying) {
              _controller.pause();
            } else {
              // If the video is paused, play it.
              _controller.play();
            }
          });
        },
        // Display the correct icon depending on the state of the player.
        child: Icon(
          _controller.value.isPlaying ? Icons.pause : Icons.play_arrow,
        ),
      ),
    );
  }
}
```

## Logs

Pasting the error instead of everything before as this was just logs of the application starting.  If you want those logs i will paste them in as well but considering this is a fresh project duplicated from the above mentioned site you can easily reproduce it.
```
[+21245 ms] W/est_videoplaye(14474): Accessing hidden method Landroid/media/AudioTrack;->getLatency()I (greylist, reflection, allowed)
[   +8 ms] I/ExoPlayerImpl(14474): Init b9f4e8e [ExoPlayerLib/2.9.6] [generic_x86, Android SDK built for x86, Google, 29]
[  +77 ms] D/NetworkSecurityConfig(14474): No Network Security Config specified, using platform default
[ +301 ms] I/OMXClient(14474): IOmx service obtained
[   +2 ms] I/ACodec  (14474): codec does not support config priority (err -2147483648)
[  +44 ms] I/ACodec  (14474): codec does not support config operating rate (err -2147483648)
[ +134 ms] I/flutter (14474): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
[        ] I/flutter (14474): The following assertion was thrown building FutureBuilder<void>(state:
[        ] I/flutter (14474): _FutureBuilderState<void>#0f0f0):
[        ] I/flutter (14474): 'package:flutter/src/rendering/proxy_box.dart': Failed assertion: line 398 pos 15: 'aspectRatio >
[        ] I/flutter (14474): 0.0': is not true.
[  +27 ms] I/flutter (14474): 
[        ] I/flutter (14474): Either the assertion indicates an error in the framework itself, or we should provide substantially
[        ] I/flutter (14474): more information in this error message to help you determine and fix the underlying cause.
[        ] I/flutter (14474): In either case, please report this assertion by filing a bug on GitHub:
[        ] I/flutter (14474):   https://github.com/flutter/flutter/issues/new?template=BUG.md
[        ] I/flutter (14474): 
[        ] I/flutter (14474): Widget creation tracking is currently disabled. Enabling it enables improved error messages. It can
[        ] I/flutter (14474): be enabled by passing `--track-widget-creation` to `flutter run` or `flutter test`.
[        ] I/flutter (14474): 
[        ] I/flutter (14474): When the exception was thrown, this was the stack:
[        ] I/flutter (14474): #2      new RenderAspectRatio (package:flutter/src/rendering/proxy_box.dart:398:15)
[        ] I/flutter (14474): #3      AspectRatio.createRenderObject (package:flutter/src/widgets/basic.dart:2673:65)
[        ] I/flutter (14474): #4      RenderObjectElement.mount (package:flutter/src/widgets/framework.dart:4749:28)
[  +24 ms] I/flutter (14474): #5      SingleChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:5126:11)
[        ] I/flutter (14474): #6      Element.inflateWidget (package:flutter/src/widgets/framework.dart:3101:14)
[        ] I/flutter (14474): #7      Element.updateChild (package:flutter/src/widgets/framework.dart:2904:12)
[        ] I/flutter (14474): #8      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:3961:16)
[        ] I/flutter (14474): #9      Element.rebuild (package:flutter/src/widgets/framework.dart:3738:5)
[        ] I/flutter (14474): #10     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2348:33)
[  +21 ms] I/flutter (14474): #11     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:760:20)
[        ] I/flutter (14474): #12     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:280:5)
[        ] I/flutter (14474): #13     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1033:15)
[        ] I/flutter (14474): #14     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:975:9)
[        ] I/flutter (14474): #15     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:891:5)
[        ] I/flutter (14474): #19     _invoke (dart:ui/hooks.dart:249:10)
[        ] I/flutter (14474): #20     _drawFrame (dart:ui/hooks.dart:207:3)
[        ] I/flutter (14474): (elided 5 frames from class _AssertionError and package dart:async)
[        ] I/flutter (14474): 
[        ] I/flutter (14474): ════════════════════════════════════════════════════════════════════════════════════════════════════
```

```
Analyzing test_videoplayer...                                           
No issues found! (ran in 13.1s)
```

```
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version 10.0.18362.418], locale en-CA)
    • Flutter version 1.9.1+hotfix.6 at C:\flutter
    • Framework revision 68587a0916 (7 weeks ago), 2019-09-13 19:46:58 -0700
    • Engine revision b863200c37
    • Dart version 2.5.0


[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\serbs\AppData\Local\Android\Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_SDK_ROOT = C:\Users\serbs\AppData\Local\Android\Sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)
    • All Android licenses accepted.

[!] Android Studio (version 3.5)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b03)

[√] VS Code (version 1.39.2)
    • VS Code at C:\Users\serbs\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.5.1

[√] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 1 category.
```
Let me know if there is anything needed.  Thank you!

**Comment by milano23** [comment]
if i change out the AspectRatio widget to a Container with a fixed size like in the code below:
```
child: Container(
                height: 225.0,
                width: 400.0,
                child: VideoPlayer(_controller),
              ),
```
I get the following error:
```
I/ExoPlayerImpl(18643): Release 2063ecb [ExoPlayerLib/2.9.6] [generic_x86, Android SDK built for x86, Google, 29] [goog.exo.core]
I/ExoPlayerImpl(18643): Init cfd2e8b [ExoPlayerLib/2.9.6] [generic_x86, Android SDK built for x86, Google, 29]
I/OMXClient(18643): IOmx service obtained
I/ACodec  (18643): codec does not support config priority (err -2147483648)
I/ACodec  (18643): codec does not support config operating rate (err -2147483648)
D/AudioTrack(18643): getTimestamp_l(57): device stall time corrected using current time 25395966228840
W/AudioTrack(18643): getTimestamp_l(57): retrograde timestamp time corrected, 25395995292240 < 25395999207340
D/AudioTrack(18643): stop(57): called with 361472 frames delivered
```

**Comment by milano23** [comment]
I have confirmed the video player is still working on my iPhone, unfortunately i am unable to test on a real world android device.  The android simulator had worked up until last night.  I have noticed on the video player plugin site it does say the ios simulator wont work.

**Comment by BondarenkoStas** [comment]
@milano23 Thank you for the report! I can run your example (and the example from flutter.dev) on the same stable and android 10 emulator without crashes, but with having logs 
```
I/ExoPlayerImpl( 5274): Release 271b46e [ExoPlayerLib/2.9.6] [generic_x86_64, Android SDK built for x86_64, Google, 29] [goog.exo.core]
D/SurfaceUtils( 5274): disconnecting from surface 0x76c584933010, reason disconnectFromSurface
I/ExoPlayerImpl( 5274): Init 2a87f2e [ExoPlayerLib/2.9.6] [generic_x86_64, Android SDK built for x86_64, Google, 29]
I/OMXClient( 5274): IOmx service obtained
D/SurfaceUtils( 5274): connecting to surface 0x76c5d4b30010, reason connectToSurface
I/MediaCodec( 5274): [OMX.google.h264.decoder] setting surface generation to 5400578
D/SurfaceUtils( 5274): disconnecting from surface 0x76c5d4b30010, reason connectToSurface(reconnect)
D/SurfaceUtils( 5274): connecting to surface 0x76c5d4b30010, reason connectToSurface(reconnect)
E/ACodec  ( 5274): [OMX.google.h264.decoder] setPortMode on output to DynamicANWBuffer failed w/ err -1010
I/ACodec  ( 5274): codec does not support config priority (err -1010)
I/OMXClient( 5274): IOmx service obtained
D/MediaCodec( 5274): [OMX.google.h264.decoder] setting dataspace on output surface to #104
I/ACodec  ( 5274): codec does not support config priority (err -2147483648)
I/ACodec  ( 5274): codec does not support config operating rate (err -2147483648)
D/SoftwareRenderer( 5274): setting dataspace on output surface to #104
D/AudioTrack( 5274): getTimestamp_l(10): device stall time corrected using current time 198196082487
```
Interesting that I don't get that `stop(57)` and the app continue to work just fine.

**Comment by milano23** [comment]
@BondarenkoStas Hello, thank you for looking into this.  The two emulators i have tried are Pixel 3 API29 and Android Accelerated x86 Oreo.  The Oreo one has always given me issues but the Pixel 3 emulator has worked fine up until now.  I have been coding with this emulator for a couple weeks with the video player plugin but only recently had issue.  I have downloaded the Pixel 3a XL emulator and can confirm it is working.  Strange that both other emulators would stop working.  Again thanks for looking into this.  Looks like there is no issue after all.
